src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +152,1305 @@ static DamageTable D_80B65044 = {
+            return;
+        }
+        if (limbByte == 4) {
+            SysMatrix_GetStateTranslationAndScaledY(-2000.0f, &this->limbPos[limbByte]);
+            return;
+        }
+
+        v2ptr = &this->limbPos[limbByte + 1];
+        v1ptr = D_80B65084;
+        SysMatrix_GetStateTranslationAndScaledX(-4000.0f, &this->limbPos[limbByte]);
+
+        for (i = limbByte + 1; i < 9; i++) {
+            SysMatrix_MultiplyVector3fByState(v1ptr, v2ptr);
+            v2ptr++;
+            v1ptr++;
+        }
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+// one of the two main draws
+void EnBigpo_Draw1(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    Gfx* dispHead;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if ((this->mainColor.a == 0xFF) || (this->mainColor.a == 0)) {
+        // fully visible OR fully transparent
+        dispHead = POLY_OPA_DISP;
+        gSPDisplayList(dispHead, &sSetupDL[6 * 0x19]);
+        gSPSegment(dispHead + 1, 0x0C, &D_801AEFA0);
+        gSPSegment(dispHead + 2, 0x08,
+                   Gfx_EnvColor(globalCtx->state.gfxCtx, this->mainColor.r, this->mainColor.g, this->mainColor.b,
+                                this->mainColor.a));
+        POLY_OPA_DISP = SkelAnime_Draw2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                                        EnBigpo_OverrideLimbDraw2, EnBigpo_PostLimbDraw2, &this->actor, dispHead + 3);
+
+    } else {
+        dispHead = POLY_XLU_DISP;
+        gSPDisplayList(dispHead, &sSetupDL[6 * 0x19]);
+        gSPSegment(dispHead + 1, 0x0C, &D_801AEF88);
+        gSPSegment(dispHead + 2, 0x08,
+                   Gfx_EnvColor(globalCtx->state.gfxCtx, this->mainColor.r, this->mainColor.g, this->mainColor.b,
+                                this->mainColor.a));
+        POLY_XLU_DISP = SkelAnime_Draw2(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl,
+                                        EnBigpo_OverrideLimbDraw2, EnBigpo_PostLimbDraw2, &this->actor, dispHead + 3);
+    }
+
+    func_800BE680(globalCtx, &this->actor, this->limbPos, 9, this->actor.scale.x * 71.428566f * this->unk220, 0,
+                  this->unk21C, 0x14);
+
+    SysMatrix_SetCurrentState(&this->drawMtxF);
+    EnBigpo_DrawLantern(&this->actor, globalCtx);
+    if (this->actionFunc == EnBigpo_SpawnPoCutscene6) {
+        // call the other unk draw func
+        EnBigpo_Draw4(&this->actor, globalCtx);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnBigpo_DrawScoopSoul(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0,
+                                (globalCtx->gameplayFrames * -0xF) & 0x1FF, 0x20, 0x80));
```

Comment:
```suggestion
               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0,
                                (globalCtx->gameplayFrames * -16) % 512, 0x20, 0x80));
```
Would the ` % 512` work?

Reply:
-15 instead of 16 but yes that works